### PR TITLE
plugins/crates: rename from crates-nvim to crates, move to by-name, migrate to mkNeovimPlugin

### LINKED ISSUE #1149: [PLUGIN REQUEST] crates.nvim
| Field    | Description |
|----------|-------------|
| Plugin   | `crates.nvim`  | 
| Homepage | `https://github.com/Saecki/crates.nvim`     | 
| Nixpkgs  | `vimPlugins.crates-nvim` | 

## Extra Information

The plugin has been partially added already but only as a completion source for cmp. It is currently lacking the overlays that show the current versions of the crates when editing a Cargo.toml as you can see in the video on the repo page: https://github.com/Saecki/crates.nvim

### PR DESCRIPTION
**Pull request by GaetanLepage**
#1205 will have to be rebases on this.

Fixes #1149

### CODE REVIEW

File: plugins/by-name/crates/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    lib.nixvim.mkSettingsRenamedOptionModules [ "plugins" "crates-nvim" ]
+      [ "plugins" "crates" ]
+      [
+        "enable"
+        "package"
+        {
+          old = "extraOptions";
+          new = "settings";
+        }
+      ];
```
Comment (MattSturgeon) [inline plugins/by-name/crates/default.nix:22]: This can be simplified to use `optionsRenamedToSettings`, right?

Comment (MattSturgeon) [reply plugins/by-name/crates/default.nix:22]: Oh, nevermind. I missed that it's doing the plugin rename

Comment (MattSturgeon) [reply plugins/by-name/crates/default.nix:22]: I'm not sure ottomh if this works. Doesn't `mkSettingsRenamedOptionModules` automatically nest into `settings`?

Comment (GaetanLepage) [reply plugins/by-name/crates/default.nix:22]: No, I need to also rename `enable` and `package` . This is one level upper.

Comment (GaetanLepage) [reply plugins/by-name/crates/default.nix:22]: > I'm not sure ottomh if this works. Doesn't `mkSettingsRenamedOptionModules` automatically nest into `settings`?

No. It takes two paths `old` and `new. Then, it does all the renames using the provided list of sub-paths, each being relative to `old` (for the rename source) and `new` (for the rename destination).

**Review by MattSturgeon — APPROVED** [review]

**Comment by GaetanLepage** [comment]
@mergify queue

**Comment by MattSturgeon** [comment]
> #1205 will have to be rebases on this.

I'm confused. Isn't that PR "adding" the same plugin this PR is dealing with?

**Comment by GaetanLepage** [comment]
> I'm confused. Isn't that PR "adding" the same plugin this PR is dealing with?

You are right. I was meaning that there could be a follow up to add option declarations, LSP integration...

**Pull request merged**